src/core/thread/router_table.cpp

```diff
@@ -202,9 +215,18 @@ Router *RouterTable::Allocate(void)
     Router *rval         = nullptr;
     uint8_t numAvailable = 0;
     uint8_t freeBit;
+#if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
```

Comment:
`RouterTable::Allocate` change SGTM. I tend to think `RouterTable::Allocate` should be the only method that is changed in RouterTable.